ui/apps/platform/src/hooks/useTableSelection.ts

```diff
@@ -32,7 +32,7 @@ function useTableSelection<T extends Base>(
 
     React.useEffect(() => {
         setSelected(data.map(preSelectedFunc));
-    }, [data]);
+    }, [data, preSelectedFunc]);
```

Comment:
A question about `preSelectedFunc` optional argument for my understanding.

Does the following have any potential benefit to minimize unneeded re-rendering?
* replace anonymous arrow function, for which every instantiation has different referential identity
* with reference to function assigned in module scope, which never changes

```ts
const preSelectedFuncDefault = () => false;

function useTableSelection<T extends Base>(
    data: T[],
    // determines whether value should be pre-selected or not
    preSelectedFunc: (T) => boolean = preSelectedFuncDefault
): UseTableSelection {
    // and so on
```

Only 1 out of 6 occurrences call with optional argument, the other 5 depend on default value.